src/v/kafka/server/connection_context.cc

```diff
@@ -68,9 +68,20 @@ ss::future<> connection_context::process_one_request() {
                       _rs.probe().header_corrupted();
                       return ss::make_ready_future<>();
                   }
-                  return handle_mtls_auth().then(
-                    [this, h = std::move(h.value()), s]() mutable {
+                  return handle_mtls_auth()
+                    .then([this, h = std::move(h.value()), s]() mutable {
                         return dispatch_method_once(std::move(h), s);
+                    })
+                    .handle_exception_type([this](const std::bad_alloc&) {
```

Comment:
> kafka: avoid an exceptional future ignored on bad_alloc

@jcsp are you using (ignored|abandoned) exceptional future in a loose way? because this doesn't look like such a case. any exceptional future here bubbles up and is caught in server/protocol.cc--not ignored/abandoned.